### DISCUSSION TITLE
Argon - Hydrogen

### DISCUSSION THREAD

**Discussion by green14-prog** [mutationpp/Mutationpp#275] [Q&A]
Hello Mutation++ community,

I have a question about using the Mutation++ library to calculate transport properties, chemical reaction rates, and compositions for a high-temperature gas containing hydrogen, silicon, and argon species. I’d appreciate any guidance you could provide on a few points in terms of implementation

1) I noticed the library includes argon as a species but doesn’t have a corresponding reaction mechanism. Is there a reaction mechanism available (like xml file ?) for an argon-involved mixture that I could incorporate?

2) If my mixture includes a reaction, such as A+B→C but the rate data I have does not follow a classical Arrhenius form (unlike the implementation for air_11), how could I incorporate this rate into the Mutation++ library? Does it necessarily need to follow a classical Arrhenius expression? If not, what should I do ?

3) Regarding hydrogen species, do you have any available mechanisms for a hydrogen gas mixture that includes H2, H2+, H, H+ electrons  ?

Thank you for your assistance.
Yours sincerely,

**Comment by grgbellasvki** [comment]
Hello,

For thermodynamics, NASA-9 or NASA-7 databases should be complete. For transport properties, I suggest you look at these two papers:
https://doi.org/10.1063/1.3495980
https://doi.org/10.1063/5.0098060

1. & 3. It's hard to provide chemical mechanisms with the library. What you see there are only examples. For Hydrogen you could refer to literature for modeling entry in gas giants. Unfortunately, chemical mechanisms are often very case specific.
2. Currently, only Arrhenius is supported, but it is possible to extend it to other rate laws. There was a fork in that direction, but it was never merged.

**Reply by green14-prog** [reply]
Hello grgbellasvki,

Thank you for your answers and the papers you provided. It helped a lot :) 

Regarding 2., do you have some example(s) on how to extend to other rate laws ? How can I do that ?

Yours sincerely
